### DISCUSSION TITLE
cart rule for free delivery applying on all cart items instead of condition based items

### DISCUSSION THREAD

**Discussion by vaishaliwebkul** [bagisto/bagisto#5202] [Ideas]
# Bug report

### Issue Description
cart rule for free delivery applying on all cart items instead of condition based items

#### Bagisto version
- [X] master branch
- [ ] 1.3.1
- [ ] 1.3.0
- [ ] 1.2.0

### Steps to reproduce

1. create cart rule for free shipping based on product condition
2. add multiple item in cart, one should be match with condition and other shouldn't
3. go to checkout and select shipping

### Expected result
While selecting Flat Rate shipping amount should be deducted only for one item

### Actual result
While selecting Flat Rate, both product shipping gets $0

![Screenshot_405](https://user-images.githubusercontent.com/43210406/122788763-4bb53a80-d2d4-11eb-9952-5e8682403ea7.png)
![Screenshot_406](https://user-images.githubusercontent.com/43210406/122788800-553ea280-d2d4-11eb-9b3b-3c8bafe24e9d.png)
![Screenshot_404](https://user-images.githubusercontent.com/43210406/122788793-5374df00-d2d4-11eb-8f17-a34a481a8843.png)

**Comment by amitgwachha** [comment]
waiting for the solution for this issue.

**Comment by amitgwachha** [comment]
any progress on this task?
